Cython/Utility/ModuleSetupCode.c

```diff
@@ -421,6 +421,40 @@ class __Pyx_FakeReference {
   #define __Pyx_PyThreadState_Current _PyThreadState_Current
 #endif
 
+// TSS (Thread Specific Storage) API
+#if PY_VERSION_HEX < 0x03070000 && !defined(PyThread_tss_create)
+#include "pythread.h"
+#define Py_tss_NEEDS_INIT 0
+typedef int Py_tss_t;
+static CYTHON_INLINE int PyThread_tss_create(Py_tss_t *key) {
+  *key = PyThread_create_key();
+  return 0; // PyThread_create_key reports success always
+}
+static CYTHON_INLINE Py_tss_t * PyThread_tss_alloc(void) {
+  int *key = malloc(sizeof(Py_tss_t));
```

Comment:
C++ requires a cast for the `malloc()` result. I actually think that this should use `PyObject_Malloc()` and `PyObject_Free()`  instead, which is faster for small chunks of memory.

Reply:
BTW, I think this should be `Py_tss_t *` and not `int *`. It's just cleaner.